### ISSUE TITLE
SQL: Error 'Specified key was too long; max key length is 767 bytes'

### ISSUE THREAD

**Issue by bonafideyan** [ent/ent#1378]
I don't know whether someone met this error, and how to solved it.

my db is mariadb cluster 10.1.37
log:

```
CREATE TABLE IF NOT EXISTS `ent_types`(`id` bigint unsigned AUTO_INCREMENT NOT NULL, `type` varchar(255) UNIQUE NOT NULL, PRIMARY KEY(`id`)) CHARACTER SET utf8mb4 COLLATE utf8mb4_bin
```

length(utf8mb4) * 255 > 757

**Comment by rotemtam** [comment]
Hey @bonafideyan 👋
Thanks for reporting. Can you please share a minimal schemas that reproduces the issue ? 
What version of ent and go are you using ?

**Comment by bonafideyan** [comment]
it runs well before I deploy to new environment, the new environment is mariadb 10.1.37. so I think any schemas will met the limit, right?
I has read the code, in ent/dialect/sql/schema/mysql.go 

```

// tBuilder returns the MySQL DSL query for table creation.
func (d *MySQL) tBuilder(t *Table) *sql.TableBuilder {
	b := sql.CreateTable(t.Name).IfNotExists()
	for _, c := range t.Columns {
		b.Column(d.addColumn(c))
	}
	for _, pk := range t.PrimaryKey {
		b.PrimaryKey(pk.Name)
	}
	// Charset and collation config on MySQL table.
	// These options can be overridden by the entsql annotation.
	b.Charset("utf8mb4").Collate("utf8mb4_bin")
	if t.Annotation != nil {
		if charset := t.Annotation.Charset; charset != "" {
			b.Charset(charset)
		}
		if collate := t.Annotation.Collation; collate != "" {
			b.Collate(collate)
		}
		if opts := t.Annotation.Options; opts != "" {
			b.Options(opts)
		}
	}
	return b
}
```


but I have not found where I can put the annotation to specify another charset, such as utf8.

**Comment by a8m** [comment]
Hey @bonafideyan and thanks for reporting this issue.

The `ent_types` table is created by the [`WithGlobalUniqueID`](https://entgo.io/docs/migrate/#universal-ids) option - do you use (and need) this option? If you don't need it, removing this option should solve the problem.

Now, regarding `MariaDB 10.1.37` - we don't test this version, but I checked this now and got the error. The solution (in the framework) can be to resize the column-size (`varchar(255)` -> `varchar(191)`) or by changing the table collation and charset for old versions of MariaDB.

Let me know if you want to contribute this change and I can guide you, or I can fix this in the upcoming days.

**Comment by bonafideyan** [comment]
@a8m , thank you. 
I like use global unique id feature.  as for change, how about change the charset to latin-1 or ascii, or utf8? or create an option for set the charset for it?

**Comment by a8m** [comment]
Fixed in #1379. Please update to master and update if you still have troubles with it.
